keras/src/layers/reshaping/reshape_test.py

```diff
@@ -100,6 +102,32 @@ def test_reshape_with_dynamic_batch_size_and_minus_one(self):
         reshaped = backend.compute_output_spec(layer.__call__, input)
         self.assertEqual(reshaped.shape, (None, 3, 8))
 
+    def test_reshape_layer_with_varying_input_size_and_minus_one(self):
+        input = KerasTensor((None, 6, 4))
+        layer = layers.Reshape((-1, 8))
+        layer.build(input.shape)
```

Comment:
You could remove all the calls to `build` in this test and others, it's no longer needed (just to make the code shorter).